Resync WPT for custom elements

rniwa opened the pull request:
#### 8b06b89e9a6f3665cb57d3995949cd82625163e3
<pre>
Resync WPT for custom elements
<a href="https://bugs.webkit.org/show_bug.cgi?id=249756">https://bugs.webkit.org/show_bug.cgi?id=249756</a>

Reviewed by Chris Dumez.

Resync web platform tests for custom elements as of b8cffc496b5e3ec50597efd2d38e4595b43ba521.

* LayoutTests/imported/w3c/web-platform-tests/common/gc.js: Added.
(self.garbageCollect.async if):
(self.garbageCollect.gcRec):
(self.garbageCollect):
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry-constructor-and-callbacks-are-held-strongly.html:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/ElementInternals-accessibility-expected.txt: Renamed from LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/ElementInternals-accessibility-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/ElementInternals-accessibility.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/ElementInternals-accessibility.html.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/ElementInternals-target-element-is-held-strongly.html:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/AriaMixin-element-attributes-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/AriaMixin-element-attributes.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/AriaMixin-string-attributes-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/AriaMixin-string-attributes.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/CustomElementRegistry-constructor.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/CustomElementRegistry-constructor.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/CustomElementRegistry-multi-register.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/CustomElementRegistry-multi-register.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/ShadowRoot-init-registry.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/ShadowRoot-init-registry.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/when-defined-reentry-crash.html: Added.

Canonical link: <a href="https://commits.webkit.org/258252@main">https://commits.webkit.org/258252@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3eff34aa9704f1d1ef510ce4eaf690f4ee2ee4a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101284 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10442 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34344 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110562 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/170842 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11383 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1304 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93702 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108393 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107067 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8653 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91891 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/35195 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90555 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23297 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78194 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4073 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24811 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4117 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1238 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10221 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44290 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5875 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2973 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM as long as the bots are happy.

rniwa committed:
Resync WPT for custom elements
https://bugs.webkit.org/show_bug.cgi?id=249756

Reviewed by Chris Dumez.

Resync web platform tests for custom elements as of b8cffc496b5e3ec50597efd2d38e4595b43ba521.

* LayoutTests/imported/w3c/web-platform-tests/common/gc.js: Added.
(self.garbageCollect.async if):
(self.garbageCollect.gcRec):
(self.garbageCollect):
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry-constructor-and-callbacks-are-held-strongly.html:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/ElementInternals-accessibility-expected.txt: Renamed from LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/ElementInternals-accessibility-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/ElementInternals-accessibility.html: Renamed from LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/ElementInternals-accessibility.html.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/ElementInternals-target-element-is-held-strongly.html:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/form-associated/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/AriaMixin-element-attributes-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/AriaMixin-element-attributes.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/AriaMixin-string-attributes-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/AriaMixin-string-attributes.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/reactions/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/CustomElementRegistry-constructor.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/CustomElementRegistry-constructor.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/CustomElementRegistry-multi-register.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/CustomElementRegistry-multi-register.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/ShadowRoot-init-registry.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/ShadowRoot-init-registry.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/scoped-registry/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/custom-elements/when-defined-reentry-crash.html: Added.

Canonical link: https://commits.webkit.org/258252@main
